core/protocols/aws-xml-protocol/src/main/java/software/amazon/awssdk/protocols/xml/internal/unmarshall/AwsXmlPredicatedResponseHandler.java

```diff
@@ -94,6 +95,7 @@ public AwsXmlPredicatedResponseHandler(
 
         AwsXmlUnmarshallingContext parsedResponse = parseResponse(httpResponse, executionAttributes);
         parsedResponse = decorateContextWithError.apply(parsedResponse);
+        logRequestId(httpResponse);
```

Comment:
Is this not handled by the combined response handler below?

Reply:
Yeah, it seems AwsXmlPredicatedResponseHandler is used by s3, and combined response handler is used by all other services. (I just learned it yesterday)